joomla/joomla-cms
# [4.0] Remove FTP support for com_joomlaupdate

Pull Request for Issue # .

### Summary of Changes
This PR removes FTP support for com_joomlaupdate:
- Remove **Installation method** option on Joomla update screen. We decided to drop support for FTP, so **Installation method** is now default to **Write files directly** and no need to be selected during update process.
- Remove code related to FTP from within com_joomlaupdate
- Remove  some classes from **restore.php** due to removal of FTP support.

### Testing Instructions

#### Test Upload & Update
1. Use Joomla 4 nightly build.
2. Download update package for this PR at https://ci.joomla.org/artifacts/joomla/joomla-cms/4.0-dev/33437/downloads/43198/Joomla_4.0.0-beta8-dev+pr.33437-Development-Update_Package.zip go to System -> Update --> Joomla and install that update package to apply changes from this PR to your site.
3. Download Joomla 4 nightly build update package from this link https://developer.joomla.org/nightlies/Joomla_4.0.0-beta8-dev-Development-Update_Package.zip , go to System -> Update -> Joomla , upload and install that update package, make sure update processed properly (without any error)

#### Test Live Update
1. Download update package for this PR at https://ci.joomla.org/artifacts/joomla/joomla-cms/4.0-dev/33437/downloads/43198/Joomla_4.0.0-beta8-dev+pr.33437-Development-Update_Package.zip, go to System -> Update --> Joomla and install that update package to re-apply changes from this PR to your site (because the changes were lost when you install Joomla 4 nightly build update package from previous test)
2. Login to administrator area of the site, access to System -> Update -> Joomla. Click on **Options** button in the toolbar. On the next screen, Set:

- **Update Channel** to **Custom URL**
- **Custom URL** to https://ci.joomla.org/artifacts/joomla/joomla-cms/4.0-dev/33437/downloads/43198/pr_list.xml
- Click on Save & Close button to save the change. You will be redirected back to Joomla update screen
3. Look at **Live Update**  tab. There should be a new update found. At the bottom, you will see **Install the Update**  button. Click on that button to process the update and make sure it is success.
